src/NuGet.Core/NuGet.Commands/Utility/CommandRunnerUtility.cs

```diff
@@ -106,7 +106,11 @@ public static async Task<SymbolPackageUpdateResourceV3> GetSymbolPackageUpdateRe
 
             if (packageSource == null)
             {
-                packageSource = new PackageSource(source);
+                // Since SymbolPackageUpdateResourceV3 will be requested this should be a v3 PackageSource
+                packageSource = new PackageSource(source)
+                {
+                    ProtocolVersion = 3
```

Comment:
This doesn't feel safe to me. I started CI to see if any tests fail, but NuGet's tests aren't exhaustive, so even if all tests pass, I still won't be fully confident.

Specifically, if the customer is pushing to a V2 package source, this change will force NuGet into thinking it's a V3 package source, and then this line: https://github.com/NuGet/NuGet.Client/blob/c6e9542fcc9300361f03d5bf7fbab1732d4d19bf/src/NuGet.Core/NuGet.Commands/Utility/CommandRunnerUtility.cs#L119

instead of returning null, because v2 package sources don't support this resource type, NuGet will throw an exception.

Therefore, my gut feeling is that this will break pushing packages to any V2 package source. We'll see if any tests fail.

Reply:
Actually it returns null, and does not throw an exception, that would have been preferable.

I'm ammenable to making `GetSymbolPackageUpdateResource` throw an exception if `packageSource = new PackageSource(source);` results in a V2 source.

Reply:
![image](https://github.com/NuGet/NuGet.Client/assets/295949/5d6d048e-d94e-4c7d-936a-bf8d2eaa552c)

Reply:
![image](https://github.com/NuGet/NuGet.Client/assets/295949/77411c2a-1877-4db6-a21a-8b9ac5c85fc6)

Reply:
15 func tests are failing:

- CommandLine.FuncTest.Commands.PushCommandTest.PushCommand_Server_SkipDuplicate_IsSpecified_PushProceeds
- NuGet.CommandLine.FuncTest.Commands.PushCommandTest.PushCommand_Server_SkipDuplicate_NotSpecified_PushHalts
- NuGet.CommandLine.Test.NuGetPushCommandTest.PushCommand_LogsServerWarningsWhenPresent(firstServerWarning: "First of two server warning messages", secondServerWarning: "Second of two server warning messages")
- NuGet.CommandLine.Test.NuGetPushCommandTest.PushCommand_PushToServer_GetCredentialFromPlugin
- NuGet.CommandLine.Test.NuGetPushCommandTest.PushCommand_PushToServer_DoesNotDeadLockWhenSTDOutLarge
- NuGet.CommandLine.Test.NuGetPushCommandTest.PushCommand_PushToServerBasicAuth
- NuGet.CommandLine.Test.NuGetPushCommandTest.PushCommand_APIV2Package_Endpoint
- NuGet.CommandLine.Test.NuGetPushCommandTest.PushCommand_PushToServerIntegratedWindowsAuthenticationDisableBuffering
- NuGet.CommandLine.Test.NuGetPushCommandTest.PushCommand_PushToServer
- NuGet.CommandLine.Test.NuGetPushCommandTest.PushCommand_PushTimeoutErrorMessage
- NuGet.CommandLine.Test.NuGetPushCommandTest.PushCommand_PushToServerIntegratedWindowsAuthentication
- NuGet.CommandLine.Test.NuGetPushCommandTest.PushCommand_PushToServer_ApiKeyAsThirdArgument
- NuGet.CommandLine.Test.NuGetPushCommandTest.PushCommand_PushToServerFollowRedirection
- NuGet.CommandLine.Test.NuGetPushCommandTest.PushCommand_LogsServerWarningsWhenPresent(firstServerWarning: "Single server warning message", secondServerWarning: null)
- NuGet.CommandLine.FuncTest.Commands.PushCommandTest.PushCommand_AllowsTimeoutToBeSpecifiedHigherThan100Seconds

At least two of the tests failed with timeout errors, so might just be flakey, but a bunch of them failed with HTTP 404 errors, trying to get the service index, as I suspected.

NuGet.CommandLine is nuget.exe, and is ilmerged, so only builds on Windows. It's not clear from your screenshots if you're on Windows or another OS, so it might not be easy for you to run these tests. `NuGet.CommandLine.Tests` needs a nuget.exe built, and on Mac the tests run though mono-project.org. We don't run NuGet.CommandLine.Tests on Linux. I suspect NuGet.CommandLine.FuncTests probably also need a nuget.exe binary, but I'm less familiar with that test project.

Interestingly, none of the tests in `dotnet.integration.tests` failed, which tells me we don't have any dotnet CLI tests pushing to a mock V2 server in our tests. In fact, it looks like we don't have any push tests at all 😞 It should be possibly to copy one of the nuget.exe tests and port it to use the same test infrastructure as other dotnet cli tests. But this is non-trivial for a first time contributor.

Reply:
I'm building on Windows, and ran the UnitTests provide in the solution file but did not encounter any errors there. It might be a bit difficult as you say for me to take on porting all the tests though I think it would be good to have them.

I've changed the code to only set the `ProtocolVersion` to V3 if it ends in `index.json`, following the logic I found in: https://github.com/NuGet/NuGet.Client/blob/5ce332fb834feea1f6c974d17456c8c9d5504e09/src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/TelemetryUtility.cs